### PR TITLE
TST: Update matrix

### LINKED ISSUE #173: BUG: Incompatibility with sphinx-dev
At some point this became incompatible with sphinx-dev.

Also, now quite a few versions between 6.1 and sphinx-dev are not included in the text matrix, so it would be nice to add those, too.

### PR DESCRIPTION
**Pull request by pllim**
Close #173 by updating matrix. The compatibility was already addressed in #172 .

### CODE REVIEW

**Review by bsipocz — APPROVED** [review]
Looks good. We may need to double check that the oldest version is indeed properly picked up (it's a bit concerning that it wasn't the case previously yet we got a big green checkmark)

**Comment by bsipocz** [comment]
Maybe also add a py312 to the dev testing?

File: setup.cfg [hunk 1]
```diff
@@ -28,6 +28,7 @@ test =
     pytest-cov
     cython
     coverage
+    setuptools;python_version>='3.12'
```
Comment (pllim) [inline setup.cfg:31]: @bsipocz , I needed this for the Python 3.12-dev job to pass. Does this look right or is this a symptom of something else?

**Comment by pllim** [comment]
Anyway, why don't we merge this as-is. And we can revisit Python 3.12 job if what I did here isn't the way to go. Thanks!

**Pull request merged**